hudi-spark-datasource/hudi-spark/src/test/java/org/apache/hudi/functional/TestSparkConsistentBucketClustering.java

```diff
@@ -303,16 +312,19 @@ public void testConcurrentWrite() throws IOException {
     // Concurrent is not blocked by the clustering
     writeData(HoodieActiveTimeline.createNewInstantTime(), 2000, true);
     // The records are immediately visible when the writer completes
-    Assertions.assertEquals(4000, readRecords(dataGen.getPartitionPaths()).size());
+    Assertions.assertEquals(4000, readRecords().size());
     // Clustering finished, check the number of records (there will be file group switch in the background)
     writeClient.cluster(clusteringTime, true);
-    Assertions.assertEquals(4000, readRecords(dataGen.getPartitionPaths()).size());
+    Assertions.assertEquals(4000, readRecords().size());
   }
 
-  private List<GenericRecord> readRecords(String[] partitions) {
-    return HoodieMergeOnReadTestUtils.getRecordsUsingInputFormat(hadoopConf,
-        Arrays.stream(partitions).map(p -> Paths.get(basePath, p).toString()).collect(Collectors.toList()),
-        basePath, new JobConf(hadoopConf), true, false);
+  private List<Row> readRecords() {
```

Comment:
For reviewers: Reading the data written by bulk insert row writer will have the same issue as #8838. So I use spark instead of input format to read data here.